app/src/main/java/kth/jjve/xfran/EditProfileActivity.java

```diff
@@ -0,0 +1,138 @@
+
+import com.google.android.material.navigation.NavigationView;
+
+import kth.jjve.xfran.viewmodels.UserProfileVM;
+
+public class EditProfileActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
+    private EditText mFirstName, mLastName, mWeight, mLength;
+    private TextView mEmail;
+    private DrawerLayout drawerLayout;
+    private NavigationView navigationView;
+    private Toolbar toolbar;
+
+    @SuppressLint("SetTextI18n")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_profile);
+
+        /*----- HOOKS -----*/
+        drawerLayout = findViewById(R.id.drawer_layout_editprofile);
+        navigationView = findViewById(R.id.nav_view_editprofile);
+        toolbar = findViewById(R.id.editprofile_toolbar);
+
+        mFirstName = findViewById(R.id.editProfileFirstName);
+        mLastName = findViewById(R.id.editProfileLastName);
+        mWeight = findViewById(R.id.editProfileWeight);
+        mLength = findViewById(R.id.editProfileLength);
+        mEmail = findViewById(R.id.editProfileEmail);
+        ImageButton save = findViewById(R.id.saveProfile);
+
+        /*-----  VM  -----*/
+        //Todo: check if we can somehow inherit this between editprofile and normal profile
+        UserProfileVM mUserProfileVM = ViewModelProviders.of(this).get(UserProfileVM.class);
+        mUserProfileVM.init();
+        mUserProfileVM.getUserProfile().observe(this, uP -> {
+            if (!uP.checkEmpty()){
+                mFirstName.setText(uP.getFirstName());
+                mLastName.setText(uP.getLastName());
+                mEmail.setText(uP.getEmail());
+                mWeight.setText(Double.toString(uP.getWeight()));
+                mLength.setText(Integer.toString(uP.getLength()));
+            }
+        });
+
+        /*--- OBSERVER ---*/
+        save.setOnClickListener(v -> {
+            // Since a double/int can't be directly read from a edittext,
+            // it is first read as string and then parsed
+            try {
+                String w = mWeight.getText().toString();
+                String l = mLength.getText().toString();
+                double weight = Double.parseDouble(w);
+                int length = Integer.parseInt(l);
+
+                mUserProfileVM.setUserProfile(
+                        mFirstName.getText().toString(),
+                        mLastName.getText().toString(),
+                        mEmail.getText().toString(),
+                        weight,
+                        length);
+
+                Toast.makeText(getApplicationContext(), "User profile saved", Toast.LENGTH_SHORT).show();
+                finish();
+            } catch (Exception e) {
+                e.printStackTrace();
+                Toast.makeText(getApplicationContext(), "Edittexts cannot be empty", Toast.LENGTH_SHORT).show();
+            }
```

Comment:
Same as above, what if it was simply a method in this class?

Reply:
Yep